### PR TITLE
Cache embeddings and wire up Flask

### PR DESCRIPTION
**Pull request by samschaevitz**
- Initially populate and wire up Flask configuration for a plurality of environments
- Define some basic endpoints (e.g. /embed and /classify) with placeholder functionality for now
- Define a MongoDB entity schema for Embeddings
- Wire up the local invocation of embed to write to and read from the MongoDB cache of embeddings
- Wire up /embed to the embedding impl with a cache (try with `./run server && curl -d "bert=https://tfhub.dev/google/bert_uncased_L-12_H-768_A-12/1&seq=helloworld" -X POST http://localhost:5000/embed`

### CODE REVIEW

File: app/embedder.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+        # convert numpyArray to list for storage in MongoDB
+        l_matrix = matrix.tolist()
+        e = Embedding.make(
+            dict(bert=self.bert, seq=seq, embedding=l_matrix))
+        e.m.save()
+        logging.info("Embedding matrix stored in MongoDB.")
+        return matrix
+
+    def buildEmbedding(self, seq: str):
```
Comment (bdittes) [inline app/embedder.py]: Did you mean to make this private (as in _buildEmbedding)?

File: app/embedder.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                do_lower_case=do_lower_case)
+
+    def GetEmbedding(self, seq: str):
+        # fetch from cache
+        obj = Embedding.m.get(bert=self.bert, seq=seq)
+        if obj:
+            logging.info("Using embedding matrix fetched from MongoDB...")
+            # convert list back to numpyArray
+            matrix = numpy.array(obj.embedding)
+            logging.debug(matrix)
```
Comment (bdittes) [inline app/embedder.py]: disable based on config.TESTING?

File: app/embedder.py (hunk 1 above)
Comment (samschaevitz) [reply app/embedder.py]: I did. Apparently been writing too much Go lately...

File: app/embedder.py (hunk 2 above)
Comment (samschaevitz) [reply app/embedder.py]: This is guarded by logging level already (which is configured depending on the flask environment config). Will happily reevaluate if we find this too be too noisey as is.

**Comment by samschaevitz** [comment]
$ ./run test  ..----------------------------------------------------------------------
Ran 2 tests in 1.082s

OK

 $ ./run lint
(venv) $

**Pull request merged**